src/lightkurve/targetpixelfile.py

```diff
@@ -2045,7 +2051,60 @@ def plot_pixels(
 
         return ax
 
+    @property
+    def skycatalog(self):
+        """Function that returns an astropy table of sources with the proper motion correction applied
 
+        Returns:
+        ------
+        catalog : astropy.table.Table
+        Returns an astropy table with ID, RA and Dec coordinates corrected for propermotion, and Mag
+        """
+
+        if self.mission.lower() == "kepler":
+            catalog_name = "kic"
+            radius = 4 * (np.max([self.shape[1], self.shape[2]]) + 10) / 2 * u.arcsec
+        elif self.mission.lower() == "k2":
+            catalog_name = "epic"
+            radius = 4 * (np.max([self.shape[1], self.shape[2]]) + 10) / 2 * u.arcsec
+        elif self.mission.lower() == "tess":
+            catalog_name = "tic"
+            radius = 21 * (np.max([self.shape[1], self.shape[2]]) + 10) / 2 * u.arcsec
+        else:
+            raise ValueError("Must pass a valid Target Pixel File object.")
+
+        catalog = query_skycatalog(
+            coord=SkyCoord(self.ra, self.dec, unit="deg"),
+            epoch=self.time[0],
+            radius=radius,
+            catalog=catalog_name,
+        )
+
+        column, row = self.wcs.world_to_pixel(
+            SkyCoord(catalog["RA"], catalog["DEC"], unit="deg")
+        )
+
+        catalog["Column"] = (self.column + column) * u.pix
+        catalog["Row"] = (self.row + row) * u.pix
+
+        # Cut down to targets within 1 pixel of edge of TPF
+        col_min = self.column - 1
+        row_min = self.row - 1
+
+        col_max = self.column + self.shape[2]
+        row_max = self.row + self.shape[1]
+
+        # Filter
+        catalog = catalog[
```

Comment:
Here the logic is to filter out stars outside of the tpf.

If the result is intended to be used by `tpf.interact_sky()` as well, it might not be desirable.
In `tpf.interact_sky()` use case, from time to time I'd zoom outside of the tpf, usually in cases there are some pecularity in some pixels at the edge of the tpf, I'd zoom out and see if there could be due to some stars just outside of the tpf.

Reply:
Correct the logic is to filter out stars outside of the TPF. 
One of the practical uses for this would be to recreate the scene via PRF modelling or examining nearby contamination.

The result is not intended for interact sky.